cdk/lib/content-api-firehose-client-testing.ts

```diff
@@ -0,0 +1,25 @@
+import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
+import { GuStack } from '@guardian/cdk/lib/constructs/core';
+import { GuGithubActionsRole } from '@guardian/cdk/lib/constructs/iam';
+import { GuAllowPolicy } from '@guardian/cdk/lib/constructs/iam/policies/base-policy';
+import type { App } from 'aws-cdk-lib';
+
+export class ContentApiFirehoseClientTesting extends GuStack {
+	constructor(scope: App, id: string, props: GuStackProps) {
+		super(scope, id, props);
+		new GuGithubActionsRole(this, {
+			policies: [
+				new GuAllowPolicy(this, 'kinesis-stream-access', {
+					actions: ['kinesis:DescribeStreamSummary'],
+					resources: [
+						`arn:aws:kinesis:${scope.region}:${scope.account}:stream/content-api-firehose-v2-CODE`,
```

Comment:
`scope` here is of type `App`. The [`region`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.App.html#region) and [`account`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.App.html#account) properties of `scope` will not be meaningful here as we're not using [CDK's `Stage` construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Stage.html). That is, they'll return `undefined`.

We'd want to use [`region`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Stack.html#region) and [`account`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Stack.html#account) from a [`Stack`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Stack.html), which'll utilise [CloudFormation's pseudo parameters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html).

This would become:

```ts
`arn:aws:kinesis:${this.region}:${this.account}:stream/content-api-firehose-v2-CODE`
```

Reply:
Ah brilliant! I've taken that with ffc7a328ccc1da14561dd2ef00988665ebb3b537, I can see that the new yaml produced looks like this, grabbing the `AWS::AccountId` with a ref:

```yaml
          - Action: kinesis:DescribeStreamSummary
            Effect: Allow
            Resource:
              Fn::Join:
                - ""
                - - "arn:aws:kinesis:eu-west-1:"
                  - Ref: AWS::AccountId
                  - :stream/content-api-firehose-v2-CODE
```

### Suggestion button for small edits

Separate point, just wanted to check that you know about GitHub's cool [suggestion button](https://haacked.com/archive/2019/06/03/suggested-changes/)?

![SuggestionBUtton](https://github.com/user-attachments/assets/d23c19f9-edc4-4fe8-9de5-612b440d142d)

It's not an AI thing - it just lets you, as a PR reviewer, make suggested changes within PR comments - it's pretty cool! You may already know about it (and it wouldn't have been perfect in this situation because a matching change to the `.test.ts.snap` file was also required) but I just wanted to share.

Reply:
> Separate point, just wanted to check that you know about GitHub's cool [suggestion button](https://haacked.com/archive/2019/06/03/suggested-changes/)?

I'm not a huge fan of this GitHub feature as it often breaks the idea of atomic commits. For example, I often typo when writing code in the browser, or my style doesn't match the project's linter, or the change needs to be made alongside changes to other files too, etc. When this happens we'd have additional `fix linter` commits, thus I don't tend to use the feature.

Reply:
Ah fair enough! Yeah, I think it only works well for very small edits (it does render a nice diff in the comment, which can be useful to highlight the exact change) and I think I'd often squash the resulting commits before merging a PR.